cdap-explore/src/main/java/co/cask/cdap/explore/service/hive/BaseHiveExploreService.java

```diff
@@ -913,6 +945,18 @@ public QueryStatus getStatus(QueryHandle handle) throws ExploreException, Handle
     }
   }
 
+  // helper method to throw only ExploreException and SQLException, since we know that the callables used
+  // only throw those checked exceptions
+  private <T> T doAs(NamespaceId namespaceId, Callable<T> callable) throws ExploreException, SQLException {
+    try {
+      return impersonator.doAs(namespaceId, callable);
+    } catch (ExploreException | HiveSQLException e) {
+      throw e;
+    } catch (Exception e) {
+      throw Throwables.propagate(e);
```

Comment:
I'll put a comment, but as long as we know that the callable passed in only throws `ExploreException | HiveSQLException` (besides unchecked exceptions), this will never be wrapping exceptions.